testing/integration/upgrade_test.go

```diff
@@ -162,50 +176,72 @@ func TestStandaloneUpgrade(t *testing.T) {
 		Sudo:    true, // requires Agent installation
 	})
 
-	agentFixture, err := define.NewFixture(t, define.Version())
-
-	require.NoError(t, err)
-
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
-	err = agentFixture.Prepare(ctx)
-	require.NoError(t, err, "error preparing agent fixture")
 
-	err = agentFixture.Configure(ctx, []byte(fastWatcherCfg))
-	require.NoError(t, err, "error configuring agent fixture")
+	versionList := getUpgradableVersions(ctx, t, define.Version())
+
+	for _, v := range versionList {
+		parsedVersion, err := version.ParseVersion(v)
+		if err != nil {
+			t.Logf("version %q returned by artifact api is not parseable: %v", v, err)
+			continue
+		}
+
+		if v == define.Version() || parsedVersion.IsSnapshot() {
+			t.Logf("Skipping upgrade test for version %q", v)
+			continue
+		}
+
+		t.Run(fmt.Sprintf("Upgrade %s to %s", v, define.Version()), func(t *testing.T) {
+			agentFixture, err := atesting.NewFixture(
+				t,
+				v,
+				atesting.WithFetcher(atesting.ArtifactFetcher()),
+			)
+
+			require.NoError(t, err, "error creating fixture")
+
+			err = agentFixture.Prepare(ctx)
+			require.NoError(t, err, "error preparing agent fixture")
+
+			err = agentFixture.Configure(ctx, []byte(fastWatcherCfg))
+			require.NoError(t, err, "error configuring agent fixture")
+
+			testStandaloneUpgrade(ctx, t, agentFixture, v, define.Version(), "", true)
+		})
+	}
+}
+
+func TestStandaloneUpgradeToSpecificSnapshotBuild(t *testing.T) {
+	define.Require(t, define.Requirements{
+		// Stack:   &define.Stack{},
+		Local:   false, // requires Agent installation
+		Isolate: true,
+		Sudo:    true, // requires Agent installation
+	})
 
-	const minVersionString = "8.9.0-SNAPSHOT"
-	minVersion, _ := version.ParseVersion(minVersionString)
+	minVersion := version_8_9_0_SNAPSHOT
 	pv, err := version.ParseVersion(define.Version())
 	if pv.Less(*minVersion) {
-		t.Skipf("Version %s is lower than min version %s", define.Version(), minVersionString)
+		t.Skipf("Version %s is lower than min version %s", define.Version(), minVersion)
 	}
 
-	ctx, cancel = context.WithCancel(context.Background())
-	defer cancel()
-
-	output, err := tools.InstallStandaloneAgent(agentFixture)
-	t.Logf("Agent installation output: %q", string(output))
+	// prepare the agent fixture
+	agentFixture, err := define.NewFixture(
+		t, define.Version(),
+	)
```

Comment:
Nit: to be consistent with other such occurrences in the same file.
```suggestion
	agentFixture, err := define.NewFixture(t, define.Version())
```